sparse/tests/test_core.py

```diff
@@ -197,6 +200,66 @@ def test_elemwise_binary(func, shape):
     assert_eq(func(xs, ys), func(x, y))
 
 
+@pytest.mark.parametrize('func', [
+    operator.pow, operator.truediv, operator.floordiv,
+    operator.ge, operator.le, operator.eq
+])
+@pytest.mark.filterwarnings('ignore:divide by zero')
+@pytest.mark.filterwarnings('ignore:invalid value')
+def test_auto_densification_fails(func):
+    xs = COO.from_numpy(random_x((2, 3, 4)))
+    ys = COO.from_numpy(random_x((2, 3, 4)))
+
+    with pytest.raises(ValueError):
+        func(xs, ys)
+
+
+@pytest.mark.parametrize('func, scalar', [
+    (operator.mul, 5),
+    (operator.add, 0),
+    (operator.sub, 0),
+    (operator.pow, 5),
+    (operator.truediv, 3),
+    (operator.floordiv, 4),
+    (operator.gt, 5),
+    (operator.lt, -5),
+    (operator.ne, 0),
+    (operator.ge, 5),
+    (operator.le, -3),
+    (operator.eq, 1)
+])
+def test_elemwise_scalar(func, scalar):
+    x = random_x((2, 3, 4))
+    y = scalar
+
+    xs = COO.from_numpy(x)
+
+    assert_eq(func(xs, y), func(x, y))
```

Comment:
Can we also check here that the output is a `COO` object and that the number of non-zeros hasn't significantly increased?

Reply:
It won't increase, but it can decrease. Let me add that check.